Dockerfile

```diff
@@ -1,31 +1,14 @@
-FROM ubuntu:bionic as oasis-chain-builder
+FROM rust:1.41.0 as oasis-chain-builder
 
-ENV DEBIAN_FRONTEND=noninteractive
-ENV RUSTUP_HOME=/usr/local/lib/rustup
-ENV CARGO_HOME=/usr/local/lib/cargo
+COPY . oasis-chain/
+WORKDIR oasis-chain
 
-RUN \
- apt-get update -q -q && \
- apt-get install --yes tzdata curl ca-certificates git build-essential && \
- echo 'UTC' > /etc/timezone && \
- rm /etc/localtime && \
- dpkg-reconfigure tzdata && \
- apt-get upgrade --yes && \
- curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain nightly && \
- RUSTFLAGS='-C target-feature=+aes,+ssse3' /usr/local/lib/cargo/bin/cargo install --locked --git https://github.com/oasislabs/oasis-chain.git oasis-chain && \
- cp /usr/local/lib/cargo/bin/oasis-chain /usr/local/bin/oasis-chain && \
- /usr/local/lib/cargo/bin/rustup self uninstall -y && \
- apt-get remove --purge --yes curl ca-certificates git build-essential && \
- apt autoremove --yes && \
- apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN cargo build --release --locked
 
-FROM ubuntu:bionic as oasis-chain-runner
+FROM gcr.io/distroless/cc
```

Comment:
Nice. :-)

Under which Linux user (uid) does the distroless run stuff?

Reply:
yeah, great tip. thanks!

> Under which Linux user
root, it seems. they also create a `nobody`, so I've modified the dockerfile to use that

Reply:
Awesome.